pkg/apis/core/validation/validation_test.go

```diff
@@ -13498,3 +13575,214 @@ func TestAlphaVolumePVCDataSource(t *testing.T) {
 		}
 	}
 }
+
+// helper creates a pod with name, namespace and IPs
+func makePod(podName string, podNamespace string, podIPs []core.PodIP) core.Pod {
+	return core.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: podName, Namespace: podNamespace},
+		Spec: core.PodSpec{
+			Containers: []core.Container{
+				{
+					Name: "ctr", Image: "image", ImagePullPolicy: "IfNotPresent", TerminationMessagePolicy: "File",
+				},
+			},
+			RestartPolicy: core.RestartPolicyAlways,
+			DNSPolicy:     core.DNSClusterFirst,
+		},
+		Status: core.PodStatus{
+			PodIPs: podIPs,
+		},
+	}
+}
+func TestPodIPsValidation(t *testing.T) {
+	testCases := []struct {
+		pod             core.Pod
+		enableDualStack bool
+		expectError     bool
+	}{
+		{
+			expectError:     false,
+			enableDualStack: false,
+			pod:             makePod("nil-ips", "ns", nil),
+		},
+		{
+			expectError:     false,
+			enableDualStack: false,
+			pod:             makePod("empty-podips-list", "ns", []core.PodIP{}),
+		},
+		{
+			expectError:     false,
+			enableDualStack: false,
+			pod:             makePod("single-ip-family-6", "ns", []core.PodIP{{IP: "::1"}}),
+		},
+		{
+			expectError:     false,
+			enableDualStack: false,
+			pod:             makePod("single-ip-family-4", "ns", []core.PodIP{{IP: "1.1.1.1"}}),
+		},
+		{
+			expectError:     false,
+			enableDualStack: true,
+			pod:             makePod("dual-stack-4-6", "ns", []core.PodIP{{IP: "1.1.1.1"}, {IP: "::1"}}),
```

Comment:
super nit, no action needed:  Cleaner might be to take a `ips string...` as the final argument of makePod